src/bookmarks/background/addition.js

```diff
@@ -52,13 +53,45 @@ export async function createNewPageForBookmark(id, bookmarkInfo) {
 
 export async function createBookmarkByExtension(url) {
     const pageId = generatePageDocId({ url })
-    const pageDoc = await db.get(pageId)
+    let pageDoc
+    try {
+        pageDoc = await db.get(pageId)
+    } catch (err) {
+        console.error(
+            'Error occurred while fetching page data: ',
+            err.toString(),
+        )
+    } finally {
+        const bookmarkDoc = transformToBookmarkDoc({ _id: pageId })({
+            dateAdded: Date.now(),
+            title: pageDoc.content.title,
+            url: pageDoc.url,
+        })
+        index.addPageConcurrent({ pageDoc, bookmarkDocs: [bookmarkDoc] })
+        db.put(bookmarkDoc)
+    }
+}
+
+export async function createBookmarkByTab(url, tabId) {
+    const pageId = generatePageDocId({ url })
+    let pageDoc
 
-    const bookmarkDoc = transformToBookmarkDoc({ _id: pageId })({
-        dateAdded: Date.now(),
-        title: pageDoc.content.title,
-        url: pageDoc.url,
-    })
-    index.addPageConcurrent({ pageDoc, bookmarkDocs: [bookmarkDoc] })
-    db.put(bookmarkDoc)
+    try {
+        pageDoc = await db.get(pageId)
+    } catch (err) {
```

Comment:
In this error handling code, you should check `err.status` is equal to `404`, which is Pouch's way of saying it's missing. Else you could be acting on an unrelated error. Example in [`store-page` module](https://github.com/WorldBrain/Memex/blob/master/src/page-storage/store-page.js#L18)

The nested `try...catch` is kinda weird as well. I think better to keep on a single level by putting the `await logPageVisitForBookmark` call in the upper-most `try` scope and then handle the error conditionally.

Reply:
If we will put the `await logPageVisitForBookmark` then we cannot check for the existing of pageDoc. So that I put into the other try...catch block. If we don't have pageDoc then we have to store the data first then we are making the bookmarkDoc.

Reply:
I see what you mean; we're stuck with the awkwardness of the Pouch API throwing an error if it's not found. 
To start with, the stuff in the `finally` block should be moved to a separate fn accepting `pageDoc` as input (`createBookmarkByExtension` can use it too). 
This shouldn't really be in `finally` block as pointed out with the `undefined` `pageDoc` case, so I suppose it would have to be called twice: 
- once inside the `try` block, after the `db.get`
- once in `catch` block if you use `.then` + `.catch` instead of `await` on the `logPageVisitForBookmark` call (may just want to bring all of that to another fn as well, especially if you get rid of `logPageVisitForBookmark` code)